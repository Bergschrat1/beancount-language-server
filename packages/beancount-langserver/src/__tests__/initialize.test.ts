import { initializeLspServer } from './mocks'

describe('lsp server - initialize', () => {
    it('initializes and responds to capabilities', async () => {
        const { server } = await initializeLspServer();
        expect(server.capabilities()).toMatchSnapshot()
    });

    it('registers lsp connection', async () => {
        const { connection, server } = await initializeLspServer();
        server.register(connection);

        expect(connection.onDidSaveTextDocument).toHaveBeenCalled();
        expect(connection.onDidOpenTextDocument).toHaveBeenCalled();
        expect(connection.onDocumentFormatting).toHaveBeenCalled();

    });
})
