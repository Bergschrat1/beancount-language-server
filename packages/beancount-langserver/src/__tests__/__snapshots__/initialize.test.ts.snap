// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lsp server - initializes initializes and responds to capabilities 1`] = `
Object {
  "completionProvider": Object {
    "resolveProvider": false,
    "triggerCharacters": Array [
      ":",
    ],
  },
  "definitionProvider": false,
  "documentFormattingProvider": true,
  "documentHighlightProvider": false,
  "documentSymbolProvider": false,
  "hoverProvider": false,
  "referencesProvider": false,
  "textDocumentSync": Object {
    "change": 2,
    "openClose": true,
    "save": Object {
      "includeText": false,
    },
    "willSave": false,
    "willSaveWaitUntil": false,
  },
  "workspaceSymbolProvider": false,
}
`;
